### PR TITLE
Deprecates methods related to language validation in NLS.

### PR DESCRIPTION
**Pull request by cschierle**
These will be replaced by an appropriate solution in UserManager of sirius-web (see https://github.com/scireum/sirius-web/pull/864)

Fixes: SIRI-368

### CODE REVIEW

File: src/main/java/sirius/kernel/async/CallContext.java [hunk 1]
```diff
@@ -359,6 +360,8 @@ public void removeFromMDC(String key) {
      *
      * @return a two-letter language code used for the current thread.
      */
+    @SuppressWarnings("deprecation")
+    @Explain("The replacement for NLS.getDefaultLanguage() is only available in sirius-web.")
```
Comment (andyHa) [inline src/main/java/sirius/kernel/async/CallContext.java:364]: I'd say this should remain as sirius-web will provide a lazyLanguageInstaller (?)

File: src/main/java/sirius/kernel/nls/NLS.java [hunk 2]
```diff
@@ -131,6 +131,7 @@ public static String getCurrentLang() {
      * @return the language code of the default language
      */
     @Nonnull
+    @Deprecated
     public static String getDefaultLanguage() {
```
Comment (andyHa) [inline src/main/java/sirius/kernel/nls/NLS.java:135]: shouldn't this remain?

**Review by andyHa — CHANGES_REQUESTED** [review]

File: src/main/java/sirius/kernel/nls/NLS.java (hunk 2 above)
Comment (cschierle) [reply src/main/java/sirius/kernel/nls/NLS.java:135]: Hmm... Depends on what the `defaultLanguage` actually, or rather if it is defined by the NLS or the scope settings. I'll look into this during further development

**Review by andyHa — APPROVED** [review]

**Review by sabieber — APPROVED** [review]

**Review by idlira — APPROVED** [review]

**Review by mkeckmkeck — APPROVED** [review]

**Comment by mkeckmkeck** [comment]
Potetially breaking when deprecation usage not allowed

**Pull request merged**